src/addr_families.c

```diff
@@ -24,8 +24,9 @@
 
 #define WOLFSENTRY_SOURCE_ID WOLFSENTRY_SOURCE_ID_ADDR_FAMILIES_C
 
-static int wolfsentry_addr_family_bynumber_key_cmp(struct wolfsentry_addr_family_bynumber *left, struct wolfsentry_addr_family_bynumber *right) {
-    WOLFSENTRY_RETURN_VALUE(left->number - right->number);
+static int wolfsentry_addr_family_bynumber_key_cmp(const struct wolfsentry_table_ent_header *left, const struct wolfsentry_table_ent_header *right) {
+    WOLFSENTRY_RETURN_VALUE(((const struct wolfsentry_addr_family_bynumber *)left)->number -
```

Comment:
Tracking down dereferences to number it looks like it is a uint16_t from wolfsentry_settings.h. Is there checks in place that left will 'never' be larger than right?

Looks like its set as a function pointer to .cmp_fn in addr_families.c, didn't track it farther back for where it is called.

Reply:
yup that's safe -- `wolfsentry_addr_family_insert()` checks:
```
    if ((max_addr_bits <= 0) || (max_addr_bits > WOLFSENTRY_MAX_ADDR_BITS))
        WOLFSENTRY_ERROR_RETURN(INVALID_ARG);
```
and the struct where it's stored has
```
struct wolfsentry_addr_family_bynumber {
[...]
    wolfsentry_addr_bits_t max_addr_bits;
```
and wolfsentry_settings.h has
```
typedef uint16_t wolfsentry_addr_bits_t;
```
and the limit is set up with asserts to protect it all:
```
#ifndef WOLFSENTRY_MAX_ADDR_BYTES
#define WOLFSENTRY_MAX_ADDR_BYTES 16
#elif WOLFSENTRY_MAX_ADDR_BYTES * 8 > 0xffff
#error WOLFSENTRY_MAX_ADDR_BYTES * 8 must fit in a uint16_t.
#endif

#ifndef WOLFSENTRY_MAX_ADDR_BITS
#define WOLFSENTRY_MAX_ADDR_BITS (WOLFSENTRY_MAX_ADDR_BYTES*8)
#else
#if WOLFSENTRY_MAX_ADDR_BITS > (WOLFSENTRY_MAX_ADDR_BYTES*8)
#error WOLFSENTRY_MAX_ADDR_BITS is too large for given/default WOLFSENTRY_MAX_ADDR_BYTES
#endif
#endif
```